velox/exec/RowContainer.h

```diff
@@ -526,6 +611,31 @@ class RowContainer {
     }
   }
 
+  template <typename T>
+  static void extractValuesOffsetsNoNulls(
+      const char* const* rows,
+      const BufferPtr& offsets,
+      int32_t columnOffset,
+      const vector_size_t resultOffset,
+      FlatVector<T>* result) {
+    auto numRows = offsets->size() / sizeof(vector_size_t);
+    auto offsetsVector = offsets->as<vector_size_t>();
+
+    BufferPtr valuesBuffer = result->mutableValues(resultOffset + numRows);
+    auto values = valuesBuffer->asMutableRange<T>();
+    for (int32_t i = 0; i < numRows; ++i) {
+      auto resultIndex = resultOffset + i;
+      auto row = rows[offsetsVector[i]];
```

Comment:
I wonder if we can avoid copy-pasting by creating a helper function that takes numRows and optional vector_size_t* rowNumbers pointer.

```
auto row = rows[rowNumbers ? rowNumbers[i] : i];
```

Reply:
Since this seemed like a very low level function, I was a bit hesitant going that route since that check is repeated per row.

There are some other minor differences involved also:
i) Adding a resultOffset. We could default this to 0 for the call from the older functions. But it would involve one more addition per row.
ii) Change in numRows calculation.
iii) The basic extractColumns functions also limit the result vector, whereas the new functions do not.

So all in all, the reuse didn't seem like a good idea to me.
wdyt ?